h2o-algos/src/main/java/hex/glm/GLMModel.java

```diff
@@ -1913,19 +1913,30 @@ public GLMMojoWriter getMojo() {
     return new GLMMojoWriter(this);
   }
 
-  @Override
-  protected boolean isFeatureUsedInPredict(int featureIdx) {
+  private boolean isFeatureUsedInPredict(int featureIdx, int offset){
     if (featureIdx < _output._dinfo._catOffsets.length - 1 && _output._column_types[featureIdx].equals("Enum")) {
       for (int i = _output._dinfo._catOffsets[featureIdx];
            i < _output._dinfo._catOffsets[featureIdx + 1];
            i++) {
-        if (beta()[i] != 0) return true;
+        if (beta()[i + offset] != 0) return true;
       }
       return false;
     } else {
       featureIdx += _output._dinfo._numOffsets[0] - _output._dinfo._catOffsets.length + 1;
     }
-    return beta()[featureIdx] != 0;
+    return beta()[featureIdx + offset] != 0;
+}
 
+  @Override
+  protected boolean isFeatureUsedInPredict(int featureIdx) {
+    if (_output.isMultinomialClassifier()) {
+      int outputClasses =  beta().length / _output.coefficientNames().length;
+      boolean used = false;
+      for (int i = 0; i < outputClasses; i++)
+        used |= isFeatureUsedInPredict(featureIdx, i * _output.coefficientNames().length);
```

Comment:
you can stop when you find feature is not used

Reply:
Good call @michalkurka, he can stop once he finds one coefficient that is not zero!

Reply:
Good point, thanks!